quinn/src/platform/cmsg.rs

```diff
@@ -34,8 +42,10 @@ impl<'a> Encoder<'a> {
         self.len += space;
         self.cmsg = unsafe { libc::CMSG_NXTHDR(self.hdr, cmsg).as_mut() };
     }
+}
 
-    pub fn finish(self) {
+impl<'a> Drop for Encoder<'a> {
```

Comment:
Do you really feel this is better than what you had before? It feels less idiomatic to me.

Reply:
I was concerned that the prior approach was easy to forget to do. With the new approach, borrowck forces you to ensure the `Encoder` is dropped before the buffer is used. I'm open to other ideas, though.

Reply:
I understand the concern. I think the way I would write it is to have `finish()` return an opaque newtype around the buffer which is what's actually passed to other code. Would that make sense here?

Reply:
The "other code" in this case is libc, so there's not an opportunity for a richer interface there.

Reply:
Hmm, maybe `Encoder` could take ownership of the buffer and only release it on `finish`.

Reply:
No, that won't work, it needs a stable address to be referenced in the `msghdr`.